pkg/kapp/resources/labeled_resources.go

```diff
@@ -144,7 +149,9 @@ func (a *LabeledResources) checkResourceOwnership(resources []Resource, opts All
 	var errs []error
 
 	for _, res := range resources {
-		if val, found := res.Labels()[expectedLabelKey]; found {
+		val, hasExpectedLabelKey := res.Labels()[expectedLabelKey]
+		_, hasExistsAnnotation := res.Annotations()[ExistsAnnKey]
```

Comment:
I just realised that the `res` over here is an existing resource and not a new resource. The test case is passing because the exists ann is also added to the resource deployed via kubectl. I will make relevant changes.

Reply:
Done